### PR TITLE
Document publish use feature

### LINKED ISSUE #563: [FEATURE] Multi-file Jayvee
## User Story
1. As a {Jayvee programmer}
2. I want {to organize my program into multiple files}
3. So that {I can keep an overview in larger projects}

## User Acceptance Criteria
- [ ] [RFC 0015](https://github.com/jvalue/jayvee/blob/main/rfc/0015-multi-file-jayvee/0015-multi-file-jayvee.md) is implemented
  - [ ] `publish` and `use` functionality   
  - [ ] `package` functionality 

## Examples
See [RFC 0015](https://github.com/jvalue/jayvee/blob/main/rfc/0015-multi-file-jayvee/0015-multi-file-jayvee.md).

## Definitions of Done
- [ ] A PR has been opened and accepted
- [ ] All user acceptance criteria are met
- [ ] All tests are passing

### PR DESCRIPTION
**Pull request by georg-schwarz**
Closes #563.

### CODE REVIEW

File: example/electric-vehicles.jv [hunk 1]
```diff
@@ -8,6 +8,11 @@
 // - Understand how to construct a pipeline with multiple sinks
 // - Understand the use of runtime parameters
 
+// 0. We can use elements defined in other files using the "use" syntax.
```
Comment (rhazn) [inline example/electric-vehicles.jv:11]:
```suggestion
// 0. We can use elements defined in other files with the "use" syntax.
```

File: example/state-codes.jv [hunk 2]
```diff
@@ -0,0 +1,78 @@
+// SPDX-FileCopyrightText: 2023 Friedrich-Alexander-Universitat Erlangen-Nurnberg
+//
+// SPDX-License-Identifier: AGPL-3.0-only
+
+// This file belongs to example 2: Electric Vehicles
+// Learning goals:
+// - Understand how to publish elements to use them in other files
+
+// 1. The publish keyword can be used directly when defining an element.
+publish valuetype UsStateCode oftype text {
```
Comment (rhazn) [inline example/state-codes.jv:10]: US is an abbreviation, not a word 😅 . Changing this means changing it in all comments / references as well sadly.

```suggestion
publish valuetype USStateCode oftype text {
```

File: example/state-codes.jv [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  constraints: [
+    UsStateCodeAllowlist,
+  ];
+}
+
+// 2. The publish keyword can be used after definition (see below).
```
Comment (rhazn) [inline example/state-codes.jv:16]:
```suggestion
// 2. The publish keyword can be used after a definition as well (see below).
```

File: example/state-codes.jv [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+// When using this delayed publish syntax, the published element can also be renamed for access from outside of the file.
```
Comment (rhazn) [inline example/state-codes.jv:17]:
```suggestion
// When using this delayed publish syntax, the published element can also be renamed. Doing so will make it available using the changed name whenever it is imported with the `use` syntax in another file. 
```

File: example/electric-vehicles.jv [hunk 5]
```diff
@@ -51,7 +56,7 @@ pipeline ElectricVehiclesPipeline {
       "VIN (1-10)" oftype VehicleIdentificationNumber10,
       "County" oftype text,
       "City" oftype text,
-      "State" oftype UsStateCode,
+      "State" oftype UsStateCode, // We can just use the element as if it was defined in this file.
```
Comment (rhazn) [inline example/electric-vehicles.jv:59]:
```suggestion
      "State" oftype UsStateCode, // We can just the element as if it was defined in this file.
```

**Review by rhazn — APPROVED** [review]
As usual, nitpicky comments.

File: example/electric-vehicles.jv (hunk 5 above)
Comment (georg-schwarz) [reply example/electric-vehicles.jv:59]: I don't get that suggestion :sweat_smile:

Comment (rhazn) [reply example/electric-vehicles.jv:59]: Me neither, wtf is that guy saying. I'd just ignore him.

Comment (georg-schwarz) [reply example/electric-vehicles.jv:59]: Haha :laughing:

**Pull request merged**